File: frontend/src/models/user/UserResponse.ts
export type UserResponse = {
  id: number
  username: string
  email: string
  about: string
  avatar: string
  cover: string
  isEmailVerified: boolean
  followersSize: number
  followingsSize: number
  isMyProfileSubscribed: boolean
  createdAt: Date
  updatedAt: Date
}

export type UserProfileResponse = {
  username: string
  about: string
  avatar: string
  cover: string
  followersSize: number
  followingsSize: number
  isMyProfileSubscribed: boolean
}

export type FollowResponse = {
  username: string
  avatar: string
  isMyProfileSubscribed: boolean
}

File: src/main/java/com/pwgp/blog/dto/user/UserResponse.java
package com.pwgp.blog.dto.user;

import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;

@Getter
@Setter
public class UserResponse {
    private Long id;
    private String username;
    private String email;
    private String about;
    private String avatar;
    private String cover;
//    private Set<Post> posts = new HashSet<Post>();
//    private Set<Post> viewedPosts = new HashSet<Post>();
    private Boolean isEmailVerified;
    private int followersSize;
    private int followingsSize;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
